apps/src/studio/studio.js

```diff
@@ -6983,9 +6983,11 @@ var checkFinished = function() {
   var hasSuccessCondition =
     level.goal && level.goal.successCondition ? true : false;
   var achievedOptionalSuccessCondition =
-    !hasSuccessCondition || utils.valueOr(level.goal.successCondition(), true);
+    !hasSuccessCondition ||
+    utils.valueOr(level.goal.successCondition(studioMsg), true);
```

Comment:
This seems like a simple solution to the problem you described, but I'm have trouble figure out how all these pieces fit together. Could you help me understand?

`studioMsg` is all the translated strings, right? If so, why do we need to pass all the translated strings into this function? Why can't the implementation of `successCondition` currently access our translated strings?

I also don't understand where the `successConfition` function is defined which will make use of the translated screens.

Reply:
`successCondition` is an attribute of a level that contains a small javascript function that checks for "success".

For example, [this level](https://studio.code.org/s/coursec-2017/stage/13/puzzle/4) checks that a [certain sprite says a certain phrase](https://github.com/code-dot-org/code-dot-org/blob/d1dfb23a39e686ed4d37b013759363654b10e67d/dashboard/config/scripts/levels/courseC_PlayLab_events4.level#L9).
![Screenshot 2020-06-30 at 10 20 54 AM](https://user-images.githubusercontent.com/46464143/86157070-b239bb00-babb-11ea-918f-0393b743cac2.png)

I couldn't get that function to access the translated strings on its own and I'm concerned about that being a sharp corner if we ever changed how files access the localized strings. We might forget to change these levels too.

Does that help?

Reply:
After this change, the strings can be accessed like:
![Screenshot 2020-06-30 at 10 28 40 AM](https://user-images.githubusercontent.com/46464143/86157605-823ee780-babc-11ea-864b-624a9a597626.png)